Realm.Shared/User.cs

```diff
@@ -0,0 +1,62 @@
+﻿////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+using System;
+using System.Threading.Tasks;
+
+namespace Realms
+{
+    public class User
+    {
+        public static User GetCurrentUser()
+        {
+            throw new NotImplementedException();
+        }
+
+        public static User FromJson()
```

Comment:
I would suggest we have `User` implement `ISerializable` which Json.NET and other serializers support natively.